### PR TITLE
auth still working

### PR DESCRIPTION
**Pull request by Abigail-cloud**
created the login function without the email functionality

### CODE REVIEW

File: src/Controller/Auth/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import  { Request, Response, NextFunction } from 'express';
+import AuthService from '../../Service/Auth/auth.service';
+const authStore = new AuthService();
```
Comment (forestbaba) [inline src/Controller/Auth/auth.controller.ts:3]: Change this name to AuthService instead

File: src/Models/user.model.ts [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+  id: number;
+  user_name: string;
+  email: string;
+  password: string;
+  phone_number: string;
+  country: Country;
+}
+export class User extends Model{
+  static get tableName() {
+    return 'users';
```
Comment (forestbaba) [inline src/Models/user.model.ts:15]: Declare strings like this as a constant string and import them

File: src/Service/Auth/auth.service.ts [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+import bcrypt from "../../Utils/bcrypt.encrypt"
+
+
+// instances
+const authRepository = new AuthRepository;
+const validations = new validation;
+
+
+export default class AuthService {
+    public async registerUser(req: Request, res: Response, next: NextFunction) {
```
Comment (forestbaba) [inline src/Service/Auth/auth.service.ts:20]: This function is not modular...It's doing alot at the same time. You can breake some functionality of it into some other sub functions

File: src/Service/Auth/auth.service.ts [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+            
+         } catch (error) {
+            return res.status(StatusCodes.BAD_REQUEST).json({error});
+         }
+        }
+
+//Login
+
+        public async login(req: Request, res: Response, next: NextFunction) {
```
Comment (forestbaba) [inline src/Service/Auth/auth.service.ts:75]: This function is not modular

File: src/Service/Auth/auth.service.ts [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+            const user = {
+                user_name: userName,
+                password_digest: password,
+                phone_number: phoneNumber,
+                email,
+                country: country,
+            };
+
+            const userEmail = await validations.where('email', user.email);
```
Comment (forestbaba) [inline src/Service/Auth/auth.service.ts:32]: I also see a sample code like this used in few other plaaces...What's the tradeoff with this approach?

File: src/Utils/ValidationUtil/validation.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { User, UserType } from '../../Models/user.model';
```
Comment (forestbaba) [inline src/Utils/ValidationUtil/validation.ts:1]: What major advantage does this class provides?

File: src/Service/Auth/auth.service.ts [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+} from "../../Utils/ErrorUtil"
+import bcrypt from "../../Utils/bcrypt.encrypt"
+
+
+// instances
+const authRepository = new AuthRepository;
+const validations = new validation;
+
+
+export default class AuthService {
```
Comment (forestbaba) [inline src/Service/Auth/auth.service.ts:19]: How come your service has all the access to Request and Response but your controllers are not doing anything?
For instance **const { userName, password, country, phoneNumber, email } = req.body;** Should be done inside a controller and passes down to service

**Review by forestbaba — COMMENTED** [review]
Great job by the way, but some stuffs still needs work

File: src/Service/Auth/auth.service.ts (hunk 7 above)
Comment (Abigail-cloud) [reply src/Service/Auth/auth.service.ts:19]: Oh. I will make necessaries corrections and make changes

File: src/Models/user.model.ts (hunk 2 above)
Comment (Abigail-cloud) [reply src/Models/user.model.ts:15]: In a different file?

Comment (3rdive) [reply src/Models/user.model.ts:15]: Yes

**Pull request closed without merging**